### PR TITLE
Changed legacy global googlesitekit in assets.

### PR DESCRIPTION
**Pull request by ShahAaron**
## Summary

Addresses issue #1615 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by ssimouh** [comment]
how can i find flags and edit on code javascript to bypasse the access premition

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Comment by aaemnnosttv** [comment]
@ShahAaron can you please resolve the merge conflict? Should be pretty quick 👍

File: assets/js/components/setup-module.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ class SetupModule extends Component {
 
 		let blockedByParentModule = false;
 
-		const { modules } = global.googlesitekit;
+		const { modules } = global._googlesitekitLegacyData;
```
Comment (aaemnnosttv) [inline assets/js/components/setup-module.js:100]: This looks like a replacement that was missed from a while back.
I believe it should be this:
```suggestion
		const modules = getModulesData();
```
(may need to import that utility)

File: assets/js/components/settings/settings-module.js [hunk 2]
```diff
@@ -149,7 +149,7 @@ class SettingsModule extends Component {
 	// Find modules that depend on a module.
 	getDependentModules() {
 		const { slug } = this.props;
-		const { modules } = global.googlesitekit;
+		const { modules } = global._googlesitekitLegacyData;
```
Comment (aaemnnosttv) [inline assets/js/components/settings/settings-module.js:152]:
```suggestion
		const modules = getModulesData();
```

File: assets/js/components/post-searcher.js [hunk 3]
```diff
@@ -146,7 +146,7 @@ class PostSearcher extends Component {
 	}
 
 	render() {
-		const { modules } = global.googlesitekit;
+		const { modules } = global._googlesitekitLegacyData;
```
Comment (aaemnnosttv) [inline assets/js/components/post-searcher.js:149]:
```suggestion
		const modules = getModulesData();
```

File: assets/js/components/dashboard-splash/dashboard-splash-app.js [hunk 4]
```diff
@@ -38,21 +38,21 @@ class DashboardSplashApp extends Component {
...
-		const { connectURL } = global.googlesitekit.admin;
+		const { connectURL } = global._googlesitekitLegacyData.admin;
 
 		const {
 			showModuleSetupWizard,
 			isAuthenticated,
 			isVerified,
 			hasSearchConsoleProperty,
-		} = global.googlesitekit.setup; /*eslint camelcase: 0*/
+		} = global._googlesitekitLegacyData.setup; /*eslint camelcase: 0*/
```
Comment (aaemnnosttv) [inline assets/js/components/dashboard-splash/dashboard-splash-app.js:48]: Is this eslint override still necessary?

File: .storybook/config.js [hunk 5]
```diff
@@ -77,12 +77,12 @@ wp.i18n.sprintf = sprintf || {};
...
-global.googlesitekit = global.googlesitekit || googlesitekit;
-global.googlesitekit.setup = global.googlesitekit.setup || googlesitekit.setup;
-global.googlesitekit.admin = global.googlesitekit.admin || googlesitekit.admin;
-global.googlesitekit.modules = global.googlesitekit.modules || googlesitekit.modules;
-global.googlesitekit.admin.assetsRoot = './assets/';
-global.googlesitekit.isStorybook = true;
+global._googlesitekitLegacyData = global._googlesitekitLegacyData || googlesitekit;
+global._googlesitekitLegacyData.setup = global._googlesitekitLegacyData.setup || googlesitekit.setup;
+global._googlesitekitLegacyData.admin = global._googlesitekitLegacyData.admin || googlesitekit.admin;
+global._googlesitekitLegacyData.modules = global._googlesitekitLegacyData.modules || googlesitekit.modules;
```
Comment (aaemnnosttv) [inline .storybook/config.js:83]: Let's see if we can simplify this a little bit as most of this looks unnecessary.

Can you try removing the above:
```diff
- // Default Data.
- const googlesitekit = dashboardData;
```
and assigning here?
```suggestion
global._googlesitekitLegacyData = dashboardData;
```

Comment (ShahAaron) [reply .storybook/config.js]: All updated as requested

File: assets/js/components/dashboard-splash/dashboard-splash-app.js (hunk 4 above)
Comment (ShahAaron) [reply assets/js/components/dashboard-splash/dashboard-splash-app.js]: Removed

File: assets/js/components/post-searcher.js (hunk 3 above)
Comment (ShahAaron) [reply assets/js/components/post-searcher.js]: Done

File: assets/js/components/settings/settings-module.js (hunk 2 above)
Comment (ShahAaron) [reply assets/js/components/settings/settings-module.js]: Done

File: assets/js/components/setup-module.js (hunk 1 above)
Comment (ShahAaron) [reply assets/js/components/setup-module.js]: Done

File: assets/js/components/setup.js [hunk 6]
```diff
@@ -42,17 +42,17 @@ class Setup extends Component {
...
+		const { connectURL } = global._googlesitekitLegacyData.admin;
 
 		const {
 			isAuthenticated,
 			hasSearchConsoleProperty,
 			isSiteKitConnected,
 			isVerified,
 			needReauthenticate,
-		} = global.googlesitekit.setup; /*eslint camelcase: 0*/
+		} = global._googlesitekitLegacyData.setup; /*eslint camelcase: 0*/
```
[the remark this answers is not in the document]
Comment (ShahAaron) [reply assets/js/components/setup.js]: Removed

File: assets/js/googlesitekit/api/index.js [hunk 7]
```diff
@@ -139,8 +139,8 @@ export const siteKitRequest = async ( type, identifier, datapoint, {
 		// if so and there is a data store available to dispatch on, dispatch a
 		// `setPermissionScopeError()` action.
 		// Kind of a hack, but scales to all components.
-		if ( error.code === PERMISSION_SCOPE_ERROR_CODE && global.googlesitekit?.data?.dispatch?.( CORE_USER ) ) {
-			global.googlesitekit.data.dispatch( CORE_USER ).setPermissionScopeError( error );
+		if ( error.code === PERMISSION_SCOPE_ERROR_CODE && global._googlesitekitLegacyData?.data?.dispatch?.( CORE_USER ) ) {
+			global._googlesitekitLegacyData.data.dispatch( CORE_USER ).setPermissionScopeError( error );
```
[the remark this answers is not in the document]
Comment (ShahAaron) [reply assets/js/googlesitekit/api/index.js]: Done

File: assets/js/util/index.js [hunk 8]
```diff
@@ -511,10 +511,10 @@ export const activateOrDeactivateModule = async ( restApiClient, moduleSlug, sta
  * @param {Object} settingsMapping The mapping between form settings names and saved settings.
  * @param {Object} settingsState   The changed settings component state to compare with.
  * @param {Object} skipDOM         Skip DOm checks/modifications, used for testing.
- * @param {Object}  _googlesitekit googlesitekit global; can be replaced for testing.
+ * @param {Object}  _googlesitekit _googlesitekitLegacyData global; can be replaced for testing.
  * @return {(void|boolean)} True if a module has been toggled.
  */
-export const toggleConfirmModuleSettings = ( moduleSlug, settingsMapping, settingsState, skipDOM = false, _googlesitekit = global.googlesitekit ) => {
+export const toggleConfirmModuleSettings = ( moduleSlug, settingsMapping, settingsState, skipDOM = false, _googlesitekit = global._googlesitekitLegacyData ) => {
```
[the remark this answers is not in the document]
Comment (ShahAaron) [reply assets/js/util/index.js]: Updated

File: assets/js/googlesitekit-modules.js [hunk 9]
```diff
@@ -29,7 +29,7 @@ if ( typeof global.googlesitekit === 'undefined' ) {
 
 if ( typeof global.googlesitekit.modules === 'undefined' ) {
 	global.googlesitekit.modules = Modules;
-} else { // TODO: Remove this once the old googlesitekit.modules has been phased out.
```
[the remark this answers is not in the document]
Comment (ShahAaron) [reply assets/js/googlesitekit-modules.js:32]: Removed

File: assets/js/components/dashboard-splash/dashboard-splash-app.js [hunk 10]
```diff
@@ -97,7 +97,7 @@ class DashboardSplashApp extends Component {
 		// Set webpackPublicPath on-the-fly.
 		if ( global._googlesitekitLegacyData && global._googlesitekitLegacyData.publicPath ) {
 			// eslint-disable-next-line no-undef
-			__webpack_public_path__ = global._googlesitekitLegacyData.publicPath;
+			__webpack_public_path__ = global._googlesitekitLegacyData.publicPath; /*eslint camelcase: 0*/
```
Comment (aaemnnosttv) [inline assets/js/components/dashboard-splash/dashboard-splash-app.js]: I think the reason this didn't come up before is that this eslint syntax is turning it off for the whole file (or perhaps just from that point on. We should only be making exceptions per-line.

Can you add `camelcase` after the existing exception for `no-undef` on the line above? 
i.e. `// eslint-disable-next-line no-undef camelcase` should work.

Note other lint warnings/errors may pop up from this change so you may need to add additional single exceptions for them, if any.

File: assets/js/util/index.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (aaemnnosttv) [reply assets/js/util/index.js]: Not quite 😄   As mentioned in the review above, it should have a double underscore in the signature:

```diff
- _googlesitekitLegacyData = global._googlesitekitLegacyData
+ __googlesitekitLegacyData = global._googlesitekitLegacyData
```
(with references updated accordingly of course)

[the remark this answers is not in the document]
Comment (ShahAaron) [reply assets/js/util/index.js]: Fixed

File: assets/js/components/dashboard-splash/dashboard-splash-app.js (hunk 10 above)
Comment (ShahAaron) [reply assets/js/components/dashboard-splash/dashboard-splash-app.js]: Used `// eslint-disable-next-line no-undef, camelcase` no other lint errors

File: assets/js/components/notifications/dashboard-setup-alerts.js [hunk 11]
```diff
@@ -69,7 +69,7 @@ class DashboardSetupAlerts extends Component {
 					winData.setupTitle = modulesData[ slug ].name;
 					winData.description = __( 'Here are some other services you can connect to see even more stats:', 'google-site-kit' );
 
-					winData = applyFilters( ` global.googlesitekit.SetupWinNotification-${ slug }`, winData );
+					winData = applyFilters( `googlesitekit.SetupWinNotification-${ slug }`, winData );
```
Comment (aaemnnosttv) [inline assets/js/components/notifications/dashboard-setup-alerts.js:72]: @felixarntz - unexpected bug that we found that is fixed now as part of this.

**Pull request merged**